### PR TITLE
CB-15833 Database engine version configurable from API for datahubs

### PR DESCRIPTION
**Pull request by daszabo**
See detailed description in the commit message.

### CODE REVIEW

**Review by lacikaaa — CHANGES_REQUESTED** [review]
shouldn't be there some kind of validation which are the accepted versions?

**Review by pdarvasi — APPROVED** [review]

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java [hunk 1]
```diff
@@ -11,11 +12,22 @@
     @NotNull
     private DatabaseAvailabilityType availabilityType;
 
+    @Pattern(regexp = "^(9\\.6|10|11|12|13|14)?$", message = "Not a valid database major version")
```
Comment (pdarvasi) [inline core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java:15]: Could you pls explain what's the rationale of storing only the major version over storing the full version here?

**Review by pdarvasi — APPROVED** [review]

**Review by keyki — APPROVED** [review]

**Review by keyki — APPROVED** [review]

**Review by oleewere — APPROVED** [review]

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java (hunk 1 above)
Comment (lacikaaa) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java:15]: also this regexp pattern is duplicated 3 times, I'm worried if it needs modification only 1 or 2 would be updated. Could you move this to something common?

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java (hunk 1 above)
Comment (daszabo) [reply core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/DatabaseBase.java]: In Azure only the major version can be specified, so we currently do not support sendin gin the minor version

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**